pkg/services/serviceaccounts/database/store.go

```diff
@@ -415,17 +425,12 @@ func (s *ServiceAccountsStoreImpl) CreateServiceAccountFromApikey(ctx context.Co
 		IsServiceAccount: true,
 	}
 
-	return s.sqlStore.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+	return s.sqlStore.InTransaction(outerCtx, func(ctx context.Context) error {
```

Comment:
switching the transaction to use `InTransaction` makes the rollback happen properly